app/decorators/user_decorator.rb

```diff
@@ -51,6 +51,12 @@ def active_or_pending_profile
     user.active_profile || pending_profile
   end
 
+  # This user's most recently activated profile that has also been deactivated
+  # due to a password reset, or nil if there is no such profile
+  def password_reset_profile
+    user.profiles.password_reset.order(activated_at: :desc).first
```

Comment:
Yeah, that's basically what the previous code was. Can even use the predicated defined by the enum `maybe_password_reset.password_reset?` as a shorthand for checking the deactivation reason against a string